### PR TITLE
GDB-7430_Execute query issues GDB-7421_Visual graph issues

### PR DESCRIPTION
**Pull request by boyan-tonchev**
What?

1. When step where SPARQL query is opened the popover dialog is not positioned properly.
2. Previous flow of execute SPARQL query block not working.
3. When open 'visual-graph-link-focus' and 'visual-graph-node-focus' guide steps some of the node are clickable.
4. Guide stop when last step "Previous" button is pressed of "visual-graph" guide block.
5. Node info side bar is open when click on second step "Previous" button of "visual-graph-properties" guide block.
6. Guide stop when click on some button "Previous" when previous step was "close node info sidebar"
7. When press button "Previous" on step which show a graph node of expanded graph the guide shows the previous step but graph is not collapsed.

Why?

1. There is race condition between setting of query and presentation of popover dialog.
2. Not implemented.
3. When the steps are opened except current selected node there are more nodes , the Shepherd library not disabled them.
4. The element of previous step was not visible because input field of search was not filled.
5. There was not implementation to do this.
6. There was not implementation to do this.
7. There was not implementation to do this.

How?

1. Added a few ms delay before popover dialog to be appeared.
2. Implemented flow of execute SPARQL query block not working.
3. Extended ShepherdService to support show and hide functions. Those functions will be called when a step is showing and hiding. When the steps are shown all graph nodes are selected and disabled. When the steps are hidden all graph nodes are selected and enabled.
4. Added onPreviousClick function which will fill the search input.
5. Added onPreviousClick function which will close the da info node side bar.
6. Added beforeShowPromise function on "close node info sidebar" step which will open the sidebar.
7. Added beforeShowPromise function on "visual-graph-expand" which will collapse the graph before step is show.

### CODE REVIEW

File: src/js/angular/guides/steps/complex/execute-sparql-query/plugin.js [hunk 1]
```diff
@@ -23,17 +23,35 @@ PluginRegistry.add('guide.step', [
                 }
             ];
 
+            const defaultQuery = 'select * where { \n\t?s ?p ?o .\n} limit 100 \n';
+            const queries = {};
+            queries[-1] = 'PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\nselect * where { \n\t?s ?p ?o .\n?o rdf:type ""\n} limit 100 ';
```
Comment (avataar) [inline src/js/angular/guides/steps/complex/execute-sparql-query/plugin.js:28]: Why is this needed?

Comment (boyan-tonchev) [reply src/js/angular/guides/steps/complex/execute-sparql-query/plugin.js:28]: This is because of previous flow. When the guide is on the step when first query is executed and user click "previous" button, this query is executed to cleared the result table.

**Review by avataar — APPROVED** [review]

**Comment by avataar** [comment]
Jenkins retry build

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Comment by boyan-tonchev** [comment]
Jenkins retry build
